src/Auth0Client.ts

```diff
@@ -850,16 +853,24 @@ export default class Auth0Client {
       );
     }
 
-    this.cacheManager.clear();
-    this.cookieStorage.remove('auth0.is.authenticated');
+    const postCacheClear = () => {
+      this.cookieStorage.remove('auth0.is.authenticated');
 
-    if (localOnly) {
-      return;
-    }
+      if (localOnly) {
+        return;
+      }
 
-    const url = this.buildLogoutUrl(logoutOptions);
+      const url = this.buildLogoutUrl(logoutOptions);
 
-    window.location.assign(url);
+      window.location.assign(url);
+    };
+
+    if (this.options.cache) {
```

Comment:
Perhaps we should do:

```js
const maybePromise = this.cacheManager.clear();
if (typeof maybePromise?.then === 'function') {
  return maybePromise.then(postCacheClear)
}
postCacheClear();
```

Then the advice would be `If you provide an async cache ...`

Reply:
Sounds reasonable!

Reply:
Diving into this, `cacheManager.clear` will always return a Promise; are you also suggesting removing `cacheManager.clearSync`?

Reply:
D'oh - no sorry, I forgot about clear/clearSync ignore me